maps/mountain_fortress_v3/charging_station.lua

```diff
@@ -34,61 +34,93 @@ local function draw_charging_gui(player, activate_custom_buttons)
     end
 end
 
+--- Searches nearby electric accumulators at position and drains them up to required power.
+---@param surface LuaSurface
+---@param position MapPosition
+---@param power_needs number The amount of power (capacity) you want to draw in Joules (native electricity unit)
 local function discharge_accumulators(surface, position, force, power_needs)
-    local accumulators = surface.find_entities_filtered {name = 'accumulator', force = force, position = position, radius = 13}
+    local accu_min_limit = 3000000 -- 3 MJ
+
+    local accumulators = surface.find_entities_filtered {type = 'accumulator', force = force, position = position, radius = 13}
     local power_drained = 0
     power_needs = power_needs * 1
+
     for _, accu in pairs(accumulators) do
-        if accu.valid then
-            if accu.energy > 3000000 and power_needs > 0 then
-                if power_needs >= 2000000 then
-                    power_drained = power_drained + 2000000
-                    accu.energy = accu.energy - 2000000
-                    power_needs = power_needs - 2000000
-                else
-                    power_drained = power_drained + power_needs
-                    accu.energy = accu.energy - power_needs
-                end
-            elseif power_needs <= 0 then
-                break
-            end
+        if power_needs <= 0 then
+            break
+        end
+
+        if accu.valid and accu.energy > accu_min_limit then
+            local accu_max_capacity = accu.electric_buffer_size
+            local accu_energy_available = math.min(accu.energy, accu_max_capacity) - accu_min_limit
+
+            local charge_delta = math.min(accu_energy_available, power_needs)
+
+            accu.energy = accu.energy - charge_delta
+            power_drained = power_drained + charge_delta
+            power_needs = power_needs - charge_delta
         end
     end
+
+    -- can anybody explain why the original author multiplied and divided by 1?
```

Comment:
Originally there was a global coefficient so map can make charging more or less expensive, but it wasn't really used and thus got substituted to 1 while the module got moved from raw global variables into tokens

Reply:
Cool thanks!